### PR TITLE
Ports nearcrit

### LINKED ISSUE #26: Crit Crawling
Crawling while you're in crit.

### PR DESCRIPTION
**Pull request by tortellinitony**
:cl: Tortellini Tony
add: Ported nearcrit.
/:cl:

It works, what more can I say?

Fixes #26 
Closes #280

### CODE REVIEW

File: tgstation.dme [hunk 1]
```diff
@@ -307,9 +307,9 @@
 #include "code\game\communications.dm"
 #include "code\game\data_huds.dm"
 #include "code\game\say.dm"
-#include "code\game\sound.dm"
 #include "code\game\shuttle_engines.dm"
 #include "code\game\skincmd.dm"
+#include "code\game\sound.dm"
```
Comment (JohnGinnane) [inline tgstation.dme:312]: What's up with these changes ?

**Review by JohnGinnane — CHANGES_REQUESTED** [review]
Absolute legend but don't forget about using whisper to suicide in near crit

File: tgstation.dme (hunk 1 above)
Comment (ghost) [reply tgstation.dme:312]: That is BYOND re-shuffling the DME it's fine

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -657,7 +664,7 @@ var/next_mob_id = 0
 			fall()
 		else if(ko || (!has_legs && !ignore_legs) || chokehold)
 			fall(forced = 1)
-	canmove = !(ko || resting || stunned || chokehold || buckled || (!has_legs && !ignore_legs && !has_arms))
+	canmove = !((is_nearcrit(src) ? 0 : ko) || resting || stunned || chokehold || buckled || (!has_legs && !ignore_legs && !has_arms))
```
[the remark this answers is not in the document]
Comment (ghost) [reply code/modules/mob/mob.dm:667]: `ko` is a dumb variable that basically determines if you are KO on the floor or not, nearcrit should overwrite that

[the remark this answers is not in the document]
Comment (JohnGinnane) [reply code/modules/mob/mob.dm:667]: Yes but the issue is `ko` is made up of `weakened || paralysis || stat || (status_flags & FAKEDEATH)`

As it is now it seems like going from health>0 to -50<health<0 means that even if you were unconscious, paralysed or weakened then you can still move...

**Review by JohnGinnane — APPROVED** [review]
Tony did some testing on the paralysis and it's working correctly, everything else looks good so let's go!

File: code/modules/mob/mob.dm [hunk 3]
```diff
@@ -632,7 +632,14 @@ var/next_mob_id = 0
 		return 0
 	return 1
 
-
+/proc/is_nearcrit(mob/living/M)
```
Comment (Carbonhell) [inline code/modules/mob/mob.dm:635]: Why make this a global proc instead of a libing proc...?

**Pull request merged**